### PR TITLE
✅ Improve Puppeteer support.

### PR DESCRIPTION
**Pull request by cvializ**
It's not perfect and not as highly prioritized, but it should still be maintained unless we decided to delete the code from the repo. This PR is a pre-requisite for the request interception feature.

Main changes
- Changed `SeleniumTestController#scroll` to be `scrollTo` to match the `FunctionalTestController` API
- Only run the Chrome browser tests for Puppeteer runs.
- Fix `ControllerPromise` `waitFunction` in `PuppeteerController` for falsy expected values.

5 tests are still not passing with Puppeteer: 3 amp carousel, one amp-form, one amp-script. But this is better than before. I'd like to merge this PR before fixing to move incrementally.

### CODE REVIEW

File: build-system/tasks/e2e/describes-e2e.js [hunk 1]
```diff
@@ -294,6 +290,20 @@ function describeEnv(factory) {
         });
     }
 
+    function getAllowedBrowsers() {
+      const {engine, browsers} = getConfig();
+
+      if (engine === 'puppeteer') {
```
[the remark this answers is not in the document]
Comment (cvializ) [reply build-system/tasks/e2e/describes-e2e.js]: Wow TIL. A TODO sounds good.

File: build-system/tasks/e2e/puppeteer-controller.js [hunk 2]
```diff
@@ -194,7 +207,9 @@ class PuppeteerController {
     const nodeListHandle = await frame.waitForFunction(
       (root, selector) => {
         const nodeList = root./*OK*/ querySelectorAll(selector);
-        return nodeList.length > 0 ? nodeList : null;
+        return nodeList.length > 0
+          ? Array.prototype.slice.call(nodeList)
```
[the remark this answers is not in the document]
Comment (cvializ) [reply build-system/tasks/e2e/puppeteer-controller.js]: 👍

File: build-system/tasks/e2e/puppeteer-controller.js [hunk 3]
```diff
@@ -70,7 +71,9 @@ async function waitFor(page, valueFn, args, condition, opt_mutate) {
       {timeout: DEFAULT_WAIT_TIMEOUT},
       ...args
     );
-    value = await handle.jsonValue();
+    const prop = await handle.jsonValue();
+    value = 'value' in prop ? prop.value : prop;
```
[the remark this answers is not in the document]
Comment (cvializ) [reply build-system/tasks/e2e/puppeteer-controller.js]: `jsonValue` doesn't return this format, the various getter methods in the class do, e.g. 

https://github.com/ampproject/amphtml/blob/fe4e16be2bfaeda647963ae5edca3fdc2f67493d/build-system/tasks/e2e/puppeteer-controller.js#L379-L386

[the remark this answers is not in the document]
Comment (sparhami) [reply build-system/tasks/e2e/puppeteer-controller.js]: More specifically, when we do:

```javascript
const handle = await evaluate(page, valueFn, ...args);
let value = await handle.jsonValue();
if (opt_mutate) {
  value = await opt_mutate(value);
}
```

above, why do we not need to do the same thing as here (check for `.value`) ? Both call `handle.jsonValue()`.

File: build-system/tasks/e2e/describes-e2e.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (estherkim) [reply build-system/tasks/e2e/describes-e2e.js]: :heart_eyes:

**Pull request merged**